client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java

```diff
@@ -255,19 +255,17 @@ public void start() throws MQClientException {
     }
 
     private void startScheduledTask() {
-        if (null == this.clientConfig.getNamesrvAddr()) {
```

Comment:
@shroman fetch name servers task must be running when client started so that client can update the failed namesrv by itself, we can  make some conditions to trigger it such as if wsAddr is not default